spec/controllers/wishlists_controller_spec.rb

```diff
@@ -144,6 +216,34 @@
         include_examples "redirects to the 'sign_in' page"
       end
     end
+
+    describe "GET #edit" do
+      context "with valid parameters" do
+        before(:each) { get_edit(id: wishlist_id) }
+
+        include_examples "redirects to the 'sign_in' page"
+      end
+
+      context "with invalid parameters" do
+        before(:each) { post_create(id: -1) }
+
+        include_examples "redirects to the 'sign_in' page"
+      end
```

Comment:
Have thought about this change but for me it's good now. Dividing tests into 2 main contexts "when user logged in" and vice versa seems fine for me. Or maybe I haven't understand You correctly.